### PR TITLE
k8s v1.28.2 and NFD updates

### PR DESCRIPTION
**Pull request by mythi**

### CODE REVIEW

**Comment by mythi** [comment]
@uniemimu PTAL 5cca930

**Comment by tkatila** [comment]
```
$ kubectl get node -o json | jq .metadata.labels | grep gpu.intel.com
  "gpu.intel.com/cards": "card0",
  "gpu.intel.com/device-id.0300-9a49.count": "1",
  "gpu.intel.com/device-id.0300-9a49.present": "true",
  "gpu.intel.com/gpu-numbers": "0",
  "gpu.intel.com/memory.max": "0",
  "gpu.intel.com/millicores": "1000",
  "gpu.intel.com/tiles": "1",
$ kubectl get node -o json | jq .status.allocatable
{
  "cpu": "8",
  "ephemeral-storage": "450548187210",
  "gpu.intel.com/i915": "8",
  "gpu.intel.com/i915_monitoring": "1",
  "gpu.intel.com/memory.max": "0",
  "gpu.intel.com/millicores": "1000",
  "gpu.intel.com/tiles": "1",
  "hugepages-1Gi": "0",
  "hugepages-2Mi": "0",
  "memory": "32474604Ki",
  "pods": "110"
}
```

I recall the extended resources vanished from the node labels previously. Now they seem to stay in labels. I wonder if it's by design?

**Comment by eero-t** [comment]
As an occasional project contributor... I think it would be good to add some note about how NFD is intended to be used to DEVEL.md.

**Comment by mythi** [comment]
> As an occasional project contributor... I think it would be good to add some note about how NFD is intended to be used to DEVEL.md.

@eero-t GPU plugin documentation talks about it and we also have similar steps documented as part of the operator deployment.

**Comment by eero-t** [comment]
> @eero-t GPU plugin documentation talks about it and we also have similar steps documented as part of the operator deployment.

I did not mean end user documentation of installing NFD rules (updated in this PR), but some mention in internal documentation on how (NFD) labels are supposed to be configured for the plugins, e.g. in the `DEVEL.md` "Checklist for New Device Plugins" section.

File: deployments/nfd/overlays/node-feature-rules/platform-labeling-rules.yaml [hunk 1]
```diff
@@ -4,6 +4,19 @@ metadata:
   name: intel-gpu-platform-labeling
 spec:
   rules:
+  - extendedResources:
+      gpu.intel.com/millicores: "@local.label.gpu.intel.com/millicores"
+      gpu.intel.com/memory.max: "@local.label.gpu.intel.com/memory.max"
+      gpu.intel.com/tiles: "@local.label.gpu.intel.com/tiles"
+    labels:
+      "gpu.intel.com/fractional-resources": "true"
```
[the remark this answers is not in the document]
Comment (mythi) [reply deployments/nfd/overlays/node-feature-rules/platform-labeling-rules.yaml]: dropped

File: deployments/nfd/base/nfd-worker.conf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+core:
+  featureSources:
+    - "cpu"
+    - "pci"
+    - "local"
+    - "kernel"
+  labelSources:
+   - "-cpu"
+   - "-pci"
+   - "-local"
```
[the remark this answers is not in the document]
Comment (mythi) [reply deployments/nfd/base/nfd-worker.conf]: fixed

**Comment by mythi** [comment]
> > @eero-t GPU plugin documentation talks about it and we also have similar steps documented as part of the operator deployment.
> 
> I did not mean end user documentation of installing NFD rules (updated in this PR), but some mention in internal documentation on how (NFD) labels are supposed to be configured for the plugins, e.g. in the `DEVEL.md` "Checklist for New Device Plugins" section.

Added: "+9. Plugin [`NodeFeatureRule`](deployments/nfd/overlays/node-feature-rules)s added for Node Feature Discovery labeling."

File: deployments/nfd/base/kustomization.yaml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
-- "https://github.com/kubernetes-sigs/node-feature-discovery/deployment/overlays/default?ref=v0.13.4"
+- "https://github.com/kubernetes-sigs/node-feature-discovery/deployment/overlays/default?ref=v0.14.1"
```
Comment (tkatila) [inline deployments/nfd/base/kustomization.yaml:4]: 0.14.0 is the latest available now.

File: deployments/nfd/base/nfd-worker.conf [hunk 4]
```diff
@@ -0,0 +1,10 @@
+core:
+  featureSources:
+    - "cpu"
+    - "pci"
+    - "local"
+    - "kernel"
+  labelSources:
+   - "-cpu"
+   - "-pci"
+   - "-kernel"
```
Comment (tkatila) [inline deployments/nfd/base/nfd-worker.conf:10]:
```suggestion
  labelSources:
   - "all"
   - "-cpu"
   - "-pci"
   - "-kernel"
```

Comment (tkatila) [reply deployments/nfd/base/nfd-worker.conf:10]: Without "all" there's nothing to remove.

**Review by tkatila — DISMISSED** [review]

**Review by tkatila — APPROVED** [review]

**Comment by mythi** [comment]
@hj-johannes-lee PTAL

**Pull request merged**